pkg/types/interfacetests/codec_interface_tests.go

```diff
@@ -0,0 +1,323 @@
+				actualEncoding, err := codec.Encode(ctx, items, TestItemArray2Type)
+
+				require.NoError(t, err)
+				assert.Equal(t, resp, actualEncoding)
+
+				var into [2]TestStruct
+				require.NoError(t, codec.Decode(ctx, actualEncoding, &into, TestItemArray2Type))
+				assert.Equal(t, items, into)
+			},
+		},
+		{
+			name: "Encodes and decodes an arrays with one element",
+			test: func(t *testing.T) {
+				item1 := CreateTestStruct(0, tester)
+				items := [1]TestStruct{item1}
+				req := &EncodeRequest{TestStructs: items[:], TestOn: TestItemArray1Type}
+				resp := tester.EncodeFields(t, req)
+
+				codec := tester.GetCodec(t)
+				actualEncoding, err := codec.Encode(ctx, items, TestItemArray1Type)
+
+				require.NoError(t, err)
+				assert.Equal(t, resp, actualEncoding)
+
+				var into [1]TestStruct
+				require.NoError(t, codec.Decode(ctx, actualEncoding, &into, TestItemArray1Type))
+				assert.Equal(t, items, into)
+			},
+		},
+		{
+			name: "Returns an error if type is undefined",
+			test: func(t *testing.T) {
+				item := CreateTestStruct(0, tester)
+				codec := tester.GetCodec(t)
+
+				_, err := codec.Encode(ctx, item, "NOT"+TestItemType)
+				assert.IsType(t, types.InvalidTypeError{}, err)
+
+				err = codec.Decode(ctx, []byte(""), item, "NOT"+TestItemType)
+				assert.IsType(t, types.InvalidTypeError{}, err)
+			},
+		},
+		{
+			name: "Returns an error encoding if arrays are the too small to encode",
+			test: func(t *testing.T) {
+				if !tester.IncludeArrayEncodingSizeEnforcement() {
+					return
+				}
+
+				item1 := CreateTestStruct(0, tester)
+				items := [1]TestStruct{item1}
+				codec := tester.GetCodec(t)
+
+				_, err := codec.Encode(ctx, items, TestItemArray2Type)
+				assert.IsType(t, types.WrongNumberOfElements{}, err)
+			},
+		},
+		{
+			name: "Returns an error encoding if arrays are the too large to encode",
+			test: func(t *testing.T) {
+				if !tester.IncludeArrayEncodingSizeEnforcement() {
+					return
+				}
+
+				item1 := CreateTestStruct(0, tester)
+				item2 := CreateTestStruct(1, tester)
+				items := [2]TestStruct{item1, item2}
+				codec := tester.GetCodec(t)
+
+				_, err := codec.Encode(ctx, items, TestItemArray1Type)
+				assert.IsType(t, types.WrongNumberOfElements{}, err)
+			},
+		},
+		{
+			name: "GetMaxEncodingSize returns errors for unknown types",
+			test: func(t *testing.T) {
+				cr := tester.GetCodec(t)
+				_, err := cr.GetMaxEncodingSize(ctx, 10, "not"+TestItemType)
+				assert.IsType(t, types.InvalidTypeError{}, err)
+			},
+		},
+	}
+	runTests(t, tester, tests)
+}
+
+// RunChainReaderWithStrictArgsInterfaceTest is meant to be used by codecs that don't pad
+// They can assure that the right argument size is verified.
+// Padding makes that harder/impossible to verify for come codecs.
+// However, the extra verification is nice to have when possible.
+func RunChainReaderWithStrictArgsInterfaceTest(t *testing.T, tester CodecInterfaceTester) {
+	ctx := context.Background()
```

Comment:
We have a helper that auto-links the context to the test deadline:
```suggestion
	ctx := tests.Context(t)
```